source/framework/core/src/TRestMetadata.cxx

```diff
@@ -515,8 +515,8 @@ TRestMetadata::TRestMetadata(const char* cfgFileName) : endl(this) {
 /// \brief TRestMetadata default destructor
 ///
 TRestMetadata::~TRestMetadata() {
-    if (fElementGlobal) delete fElementGlobal;
-    if (fElement) delete fElement;
+    delete fElementGlobal;
```

Comment:
why not this check? It is not bad to check if the pointer is `nullptr` then just do nothing in that case.

Reply:
> why not this check? It is not bad to check if the pointer is `nullptr` then just do nothing in that case.

Deleting a null pointer is perfectly safe, so there is not need to check this.

https://stackoverflow.com/questions/615355/is-there-any-reason-to-check-for-a-null-pointer-before-deleting

Reply:
The main problem is that delete will not remove the actual pointer address. This is something I discussed with @juanangp already in 2015. So, it is necessary to set it to null to make sure you wont access it again or you don't believe it is existing. Probably at the constructor it does nothing.